snuba/datasets/discover.py

```diff
@@ -30,18 +31,42 @@
 from snuba.query.timeseries import TimeSeriesExtension
 from snuba.request.request_settings import RequestSettings
 from snuba.util import is_condition
+from snuba.utils.metrics.backends.wrapper import MetricsWrapper
 
 EVENTS = "events"
 TRANSACTIONS = "transactions"
 
+metrics = MetricsWrapper(environment.metrics, "api.discover")
+
 
 def detect_table(
-    query: Query, events_only_columns: ColumnSet, transactions_only_columns: ColumnSet
+    query: Query,
+    events_only_columns: ColumnSet,
+    transactions_only_columns: ColumnSet,
+    track_bad_queries: bool,
 ) -> str:
     """
     Given a query, we attempt to guess whether it is better to fetch data from the
     "events" or "transactions" storage. This is going to be wrong in some cases.
     """
+    event_column = None
+    transaction_column = None
+    all_referenced_columns = query.get_all_referenced_columns()
+    for col in all_referenced_columns:
+        if events_only_columns.get(col):
+            event_column = col
+        elif transactions_only_columns.get(col):
+            transaction_column = col
+
+    if event_column is not None and transaction_column is not None:
```

Comment:
Feel free to log this as a warning with the logger. It will also show up in sentry with all the details, including the whole query, the referrer. Also it would not be sampled, metrics are.

Like this 
https://github.com/getsentry/snuba/blob/master/snuba/query/parser/__init__.py#L35